src/components/primitives/NoData.tsx

```diff
@@ -1,5 +1,8 @@
 import Typography, { TypographyProps } from '@mui/material/Typography';
+import React from 'react';
 
-export const NoData = ({ ...rest }: TypographyProps) => {
-  return <Typography {...rest}>—</Typography>;
+export const NoData = <C extends React.ElementType>(
+  props: TypographyProps<C, { component?: C }>
+) => {
+  return <Typography {...props}>—</Typography>;
```

Comment:
using this type for wrapping `material-ui` components  we can use the `component` prop and the it will inherit all the props from the element used as component.
https://mui.com/material-ui/guides/composition/#with-typescript